analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -29,19 +29,18 @@ namespace SonarAnalyzer.Rules
     public abstract class MethodsShouldNotHaveIdenticalImplementationsBase<TMethodDeclarationSyntax, TLanguageKindEnum> : SonarDiagnosticAnalyzer
         where TLanguageKindEnum : struct
     {
-        protected const string DiagnosticId = "S4144";
+        private const string DiagnosticId = "S4144";
         private const string MessageFormat = "Update this method so that its implementation is not identical to '{0}'.";
 
         private readonly DiagnosticDescriptor rule;
         protected abstract ILanguageFacade<TLanguageKindEnum> Language { get; }
         protected abstract TLanguageKindEnum[] SyntaxKinds { get; }
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
 
         protected abstract IEnumerable<TMethodDeclarationSyntax> GetMethodDeclarations(SyntaxNode node);
         protected abstract SyntaxToken GetMethodIdentifier(TMethodDeclarationSyntax method);
         protected abstract bool AreDuplicates(TMethodDeclarationSyntax firstMethod, TMethodDeclarationSyntax secondMethod);
-        protected virtual bool IsExcludedFromBeingExamined(ISymbol nodeContainingSymbol) =>
-            nodeContainingSymbol.Kind != SymbolKind.NamedType;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
```

Comment:
You can inherit `SonarDiagnosticAnalyzer<TSyntaxKind>` instead to remove quite a lot of this scaffolding... (and rename `TLangaugeKindEnum` to `TSynaxKind`.